script.lua

```diff
@@ -7430,8 +7486,8 @@ function swapPlayerAreaObjects(a, b, colorA, colorB)
             end
             for _,obj in ipairs(getObjects()) do
                 local objPos = obj.getPosition()
-                local powerZonePos = Player[color].getHandTransform().position + tableOffset
-                if obj.type == "Fog" and obj.getData().FogColor == color and objPos.x == powerZonePos.x and objPos.z <= powerZonePos.z then
+                local powerZonePos = playerTables[color].getPosition() + tableOffset
```

Comment:
what needed this change to happen?

Reply:
This is something that should have been changed as part of 2e05b1572b341bb0ae40302220e7a3307583daf9. Perhaps I should have submitted a separate PR for this ('cause it's a bugfix rather than part of the new feature), but I explained it in the commit message for aa35d752da51c5570b044944a4bb9f266873563b, so the explanation will still be linked from a blame, at least.